tools/check-metadata.py

```diff
@@ -0,0 +1,155 @@
+        if not sorted(props[1:]) == props[1:]:
+            return False
+
+
+def check_files(metadata, check_media_integrity=False):
+    valid = True
+    for sound in metadata:
+        location = metadata[sound].get("sound-file")
+        if not location:
+            locations = metadata[sound].get("sound-files")
+        else:
+            locations = [location]
+
+        for location in locations:
+            full_location = os.path.abspath(os.path.join(SOUNDS_DIR, location))
+            file_extension = os.path.splitext(full_location)[1]
+
+            if not os.path.isfile(full_location):
+                valid = False
+
+                print(f"File '{full_location}' does not exist.",
+                      file=sys.stderr)
+            elif (check_media_integrity and
+                    not check_media_file_integrity(full_location)):
+                valid = False
+                print(f"File '{full_location}' is not a valid media file.",
+                      file=sys.stderr)
+            if file_extension != ".webm":
+                print(f"File '{full_location}' is not a 'webm' file. "
+                      "Did you forget to convert it? "
+                      "Use ./tools/convert-sounds.sh", file=sys.stderr)
+    return valid
+
+
+def check_schema():
+    proccess = subprocess.Popen(
+        ["jsonschema", "-i", f"{args.path.name}", f"{METADATA_SCHEMA_PATH}"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    out, err = proccess.communicate()
+    is_valid = proccess.returncode == 0
+
+    if not is_valid:
+        print(err.decode("utf-8"), file=sys.stderr)
+    return is_valid
+
+
+def check_json():
+    proccess = subprocess.Popen(
+        ["python", "-m", "json.tool", f"{args.path.name}"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    out, err = proccess.communicate()
+    is_valid = proccess.returncode == 0
+
+    if not is_valid:
+        print(err.decode("utf-8"), file=sys.stderr)
+    return is_valid
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--path",
+                        type=argparse.FileType("r+"),
+                        help="Path to the metadata json file",
+                        default=SYSTEM_METADATA_PATH,
+                        required=False)
+    parser.add_argument("--check-sorted", action="store_true",
+                        help="Checks that the metadata json file is sorted.",
+                        required=False)
+    parser.add_argument("--check-files", action="store_true",
+                        help="Checks the existence of specified audio files.",
+                        required=False)
+    parser.add_argument("--check-integrity", action="store_true",
+                        help="Checks for the integrity of audio files.",
+                        required=False)
+
+    args = parser.parse_args()
+
+    if not check_json():
```

Comment:
I hope this is not necessary and the schema check would bomb out if the JSON file was invalid.

Reply:
Yes, the schema check with jsonschema exits with a non-zero status if the json file is invalid. However, this check tells you what line or where the JSON file is wrong like what https://jsonlint.com/ can do. I find it useful for this reason. Is it a good reason for you to keep this=

Reply:
OK, I see your point. You could implement it by adding another line in the CI script that runs `cat metadata.json | python3 -m json.tool > /dev/null` to get the more detailed error messages.